src/lua/CommunityBalanceMod/FortressPvE/Stormcloud/GUIAlienHUD.lua

```diff
@@ -0,0 +1,1227 @@
+                    
+                        GUI.DestroyItem(self.babblerIcons[#self.babblerIcons])
+                        table.remove(self.babblerIcons, #self.babblerIcons)    
+                
+                    end
+                
+                end
+
+                local totalIconSize = 0
+                for j = 1, #self.babblerIcons do
+
+                    local size = kBabblerIconSize
+                    local babblerIcon = self.babblerIcons[j]
+
+                    if babblerIcon:GetSize().x - kBabblerIconSize > 0.001 then
+                        size = Slerp(babblerIcon:GetSize().x, kBabblerIconSize, deltaTime * kBabblerIconSlerpSpeed)
+                        babblerIcon:SetSize(Vector(size, size, 0))
+                        babblerIcon:SetPosition(Vector(j * size, 0, 0))
+                    end
+
+                    babblerIcon:SetColor(j <= numBabblersClinged and kBabblerDefaultColor or kBabblerAlternateColor)
+                    totalIconSize = totalIconSize + size
+                
+                end
+                
+                local size = Vector(totalIconSize * numBabblersTotal, totalIconSize, 0)
+                self.babblerIndicationFrame:SetSize(size)
+                
+
+            end
+
+            function GUIAlienHUD:Update(deltaTime)
+
+                PROFILE("GUIAlienHUD:Update")
+
+                local newHudMode = Client.GetHudDetail()
+                local fullMode = newHudMode == kHUDMode.Full
+
+                if self.cachedHudDetail ~= newHudMode then
+
+                    self.cachedHudDetail = newHudMode
+                    local minimal = self.cachedHudDetail == kHUDMode.Minimal
+
+                    self.secondaryAbilityBackground:SetTexture(ConditionalValue(minimal, "ui/transparent.dds", kSmokeTexture))
+                    self.healthBall:GetBackground():SetAdditionalTexture("noise", ConditionalValue(minimal, "ui/transparent.dds", kBackgroundNoiseTexture))
+                    self.energyBall:GetBackground():SetAdditionalTexture("noise", ConditionalValue(minimal, "ui/transparent.dds", kBackgroundNoiseTexture))
+
+                    self:Reset()
+                end
+
+                -- update resource display
+                self.resourceDisplay:Update(deltaTime, { PlayerUI_GetTeamResources(), PlayerUI_GetPersonalResources(), CommanderUI_GetTeamHarvesterCount() } )
+                
+                -- updates animations
+                GUIAnimatedScript.Update(self, deltaTime)
+                
+                UpdateNotifications(self, deltaTime)
+                
+                self.inventoryDisplay:Update(deltaTime, { PlayerUI_GetActiveWeaponTechId(), PlayerUI_GetInventoryTechIds() })
+
+                -- Update player status icons
+                local playerStatusIcons = {
+                    Detected = PlayerUI_GetIsDetected(),
+                    Enzymed = PlayerUI_GetIsEnzymed(),
+Stormed = PlayerUI_GetIsStormed(),
```

Comment:
Correct me if I'm wrong, but I believe we're only changing this one line. If that's the case why are we replacing the whole file, I think it would make more sense to just replace the Update function